utils/pattern/src/frontend/mod.rs

```diff
@@ -76,134 +65,79 @@ use writeable::{adapters::TryWriteableInfallibleAsWriteable, PartsWrite, TryWrit
 /// ```
 ///
 /// [`SinglePlaceholder`]: crate::SinglePlaceholder
-#[derive(Debug, Clone, PartialEq)]
 #[cfg_attr(feature = "yoke", derive(yoke::Yokeable))]
-#[cfg_attr(
-    feature = "zerofrom",
-    derive(zerofrom::ZeroFrom),
-    zerofrom(may_borrow(Store))
-)]
-pub struct Pattern<Backend, Store: ?Sized> {
-    _backend: PhantomData<Backend>,
-    store: Store,
+#[repr(transparent)]
+pub struct Pattern<B: PatternBackend> {
+    _backend: PhantomData<B>,
+    /// The encoded storage
+    pub store: B::Store,
 }
 
-impl<Backend, Store> Pattern<Backend, Store> {
-    pub fn take_store(self) -> Store {
-        self.store
-    }
-
-    /// Creates a pattern from a serialized backing store without checking invariants.
-    /// Most users should prefer [`Pattern::try_from_store()`].
-    ///
-    /// The store is expected to come from a valid `Pattern` with this `Backend`,
-    /// such as by calling [`Pattern::take_store()`]. If the store is not valid,
-    /// unexpected behavior may occur.
-    ///
-    /// To parse a pattern string, use [`FromStr::from_str`].
-    ///
-    /// # Examples
```

Comment:
Nit: Bring back the example, please

Reply:
This is a moot examples because users don't have to do store-juggling anymore.

Reply:
Stores are still public, yes?

Reply:
No